dist/WebExtension/js/UciSimpleStorage.js

```diff
@@ -33,13 +33,13 @@ function UciStorage() {
 			this.setStoredValue(this.encode(), profilName);
 		}
 		var UsageConfortpref = this.encodeUsageConfort();
-		this.postMessage("orangeconfort+userprefsave", UsageConfortpref, document.location.protocol + '//' + document.location.hostname + document.location.pathname);
+		this.postMessage("orangeconfort+userprefsave", UsageConfortpref);
```

Comment:
Maybe there's a need to keep sending message to a dedicated host/path to avoid messages colision?

Reply:
In fact it's been 6 years since the target origin argument is not used anymore, [removed when dropping previous Firefox extension format](https://github.com/Orange-OpenSource/Orange-Confort-plus/commit/48c273bba97527b575b93beb8d4516d7c6e9de38#diff-9da9eaa42692ba25f2c104d57b9d5b19bf9c4e867bf39f125056e045d8856a53L44).

As far as I understand, since then settings are saved as a single string using the `UCI41` key, and saving them in some tab made them available in every tabs. Would worth a try.

And if so, the same goes to `isCduEnabled`, which explains the flaky behavior when toggling extension on and off.

Reply:
In fact it worked inadvertently because [extension used browser's `localStorage` at the moment](https://github.com/Orange-OpenSource/Orange-Confort-plus/blob/48c273bba97527b575b93beb8d4516d7c6e9de38/app/WebExtension/js/main.js), [which is origin specific](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) whereas [`chrome.storage.local` is not](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/set).

Still digging.

Reply:
[MDN explains](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local) why `chrome.storage.local` is recommended and enforced over `window.LocalStorage`:

> Although this API is similar to [Window.localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage), it is recommended that you don't use Window.localStorage in extension code. Firefox clears data stored by extensions using the localStorage API in various scenarios where users clear their browsing history and data for privacy reasons. Data saved using the storage.local API is correctly persisted in these scenarios.